README.md

```diff
@@ -1,223 +1,357 @@
+    + [3.14 `roles/mining-proxy`](https://github.com/stratum-mining/stratum#314--roles-mining-proxy)
+    + [3.15 `roles/pool`](https://github.com/stratum-mining/stratum#315--roles-pool)
+    + [3.16 `roles/test-utils/pool`](https://github.com/stratum-mining/stratum#316--roles-test-utils-pool)
+    + [3.17 `roles/test-utils/mining-device`](https://github.com/stratum-mining/stratum#317--roles-test-utils-mining-device)
+    + [3.18 `examples`](https://github.com/stratum-mining/stratum#318--examples)
+    + [3.19 `examples/interop-cpp`](https://github.com/stratum-mining/stratum#319--examples-interop-cpp)
+    + [3.20 `examples/interop-cpp-no-cargo`](https://github.com/stratum-mining/stratum#320--examples-interop-cpp-no-cargo)
+    + [3.21 `examples/ping-pong-with-noise`](https://github.com/stratum-mining/stratum#321--examples-ping-pong-with-noise)
+    + [3.22 `examples/ping-pong-without-noise`](https://github.com/stratum-mining/stratum#322--examples-ping-pong-without-noise)
+    + [3.23 `examples/sv1-client-and-server`](https://github.com/stratum-mining/stratum#323--examples-sv1-client-and-server)
+    + [3.24 `examples/sv2-proxy`](https://github.com/stratum-mining/stratum#324--examples-sv2-proxy)
+    + [3.25 `examples/template-provider-test`](https://github.com/stratum-mining/stratum#325--examples-template-provider-test)
+    + [3.26 `test`](https://github.com/stratum-mining/stratum#326--test)
+    + [3.27 `experimental`](https://github.com/stratum-mining/stratum#327--experimental)
+  * [4. Branches](#4-branches)
+
+## 1. Goals
+The goals of this project is to provide:
+
+1. A robust set of Stratum V2 (Sv2) primitives as Rust library crates that can be used by anyone
+   that wants to expand the protocol or implement a role (e.g. a pool wanting to support Sv2, a
+   mining-device/hashrate producer that wants to integrate Sv2 into their firmware, a Bitcoin node
+   that wants to implement Template Provider capabilities to build the `blocktemplate`).
+
+1. The above Rust primitives as a C library, making it available to non-Rust users via a C
+   bindings.
+
+1. A set of helpers built on top of the above primitives and the external Bitcoin-related Rust
+   crates that can be used by anyone who wants to implement the Sv2 roles (e.g. a pool that uses
+   Rust and wants to support Sv2).
+
+1. An open-source implementation of a Sv2 proxy that can be used by miners that want to use Sv2.
+
+1. An open-source implementation of a Sv2 pool that can be used by anyone that wants to operate a
+   Sv2 pool.
+
+
+## 2. Common Use Cases
+Different portions of the library will be used depending on the use case and/or desired
+functionality. Some examples of different use-cases are:
+
+1. A miner who runs a mining farm with Sv2-compatible mining firmware mining to a Sv2-compatible
+   pool can use this library as a proxy which allows them realize the full benefits of the Sv2
```

Comment:
change "this library" with something else.

There are 2 reasons:
1. in `this library` is not clear what `this` means
2. the proxy is not a library